### PR TITLE
#33 manage materials

### PR DESCRIPTION
**Pull request by RaidenRabit**

### CODE REVIEW

**Comment by RaidenRabit** [comment]
Just so you guys know, to run the acceptance tests, you must first run the solution, so the iis processes get initialized.

File: Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Database/CopyDa_Database.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        private readonly Context _context;
+        public CopyDa_Database(Context context)
+        {
+            _context = context;
+        }
+
+        public List<Copy> GetAvailableCopies(int isbn)
+        {
+            return _context.Copies
+                .Where(x => x.ISBN.Equals(isbn.ToString()))
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Database/CopyDa_Database.cs:19]: why do you pass isbn as int and then convert it to string?

File: Code/GeorgiaLibrarySystem-/GTLService/DataManagement/Code/CopyDm_Code.cs [hunk 2]
```diff
@@ -19,28 +19,40 @@ public int GetAvailableCopyId(int isbn)
...
-                else
+            int i = 0, id = 0;
+            if(copies.Count > 0)
+                while (!found)
                 {
-                    i++;
+                    if (_lendingDa.GetBorrow(copies[i].CopyID) == null)
+                    {
+                        found = true;
+                        id = copies[i].CopyID;
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/DataManagement/Code/CopyDm_Code.cs]: why not just return here?

**Review by bubriks — COMMENTED** [review]
does CreateCopy in CopyDa_Code work? i think it shouldnt, becouse you dont save changes after adding new element. and further more in other places as well you just automaticaly return true if no exception thrown, but that doesnt mean that info in db has been updated, example: DeleteCopy in CopyDa_Code

File: Code/Database/AppveyorScriptDatabase.bat [hunk 3]
```diff
@@ -13,3 +13,5 @@ SET PASSWORD="Password12!"
 SET INPUT=%cd%\SQLCreateQuery.sql
```
Comment (bubriks) [inline Code/Database/AppveyorScriptDatabase.bat:13]: BTW i can review compleatly all changes so some ill just have to explain

File: Code/Database/AppveyorScriptDatabase.bat [hunk 4]
```diff
@@ -13,3 +13,5 @@ SET PASSWORD="Password12!"
...
```
Comment (bubriks) [inline Code/Database/AppveyorScriptDatabase.bat:14]: ReadMaterials not used at all from what i see

File: Code/Database/AppveyorScriptDatabase.bat [hunk 5]
```diff
@@ -13,3 +13,5 @@ SET PASSWORD="Password12!"
...
 
 %SQLCMD% -S%SERVER% -d%DB% -U%LOGIN% -P%PASSWORD% -i%INPUT% -b
```
Comment (bubriks) [inline Code/Database/AppveyorScriptDatabase.bat:15]: DeleteMaterial in MaterialDa_Code does two actions (not saying it necasaraly wrong, but i thought da layer should be basic af)

File: Code/Database/AppveyorScriptDatabase.bat [hunk 6]
```diff
@@ -13,3 +13,5 @@ SET PASSWORD="Password12!"
...
+SET INPUT=%cd%\InsertScript.sql
```
Comment (bubriks) [inline Code/Database/AppveyorScriptDatabase.bat:16]: UI shows "Failed borrow Someone else was faster than you" but it failed becouse the limit was reached

File: Code/Database/AppveyorScriptDatabase.bat [hunk 7]
```diff
@@ -13,3 +13,5 @@ SET PASSWORD="Password12!"
...
+%SQLCMD% -S%SERVER% -d%DB% -U%LOGIN% -P%PASSWORD% -i%INPUT% -b
```
Comment (bubriks) [inline Code/Database/AppveyorScriptDatabase.bat:17]: Search in UI doesnt update when new type of material selected (i think it should, but then again UI not our focus so could be ok)

**Review by bubriks — APPROVED** [review]
good i guess

**Pull request merged**